app/controllers/forumthreads_controller.rb

```diff
@@ -80,6 +85,20 @@ def destroy
     redirect_to @thread.forum
   end
 
+  def search
+  end
+
+  def search_redirect
```

Comment:
> submit with a blank search box, the url will contain `?query=`

That's exactly what I'd expect, what would you not want that?

---

The `<select>` value doesn't need to match the text displayed to the user, you can set it to `""`. Other than that, that URL looks fine.